### PR TITLE
Update faq.rst for FAQ section about rendering (add examples and reformat)

### PR DESCRIPTION
**Pull request by PyExplorer**

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -258,8 +258,10 @@ https://github.com/nabilm/ansible-splash.
 
 Website is not rendered correctly
 ---------------------------------
+In this case we have two types of issues: wrong rendering and Splash crashing.
```
Comment (kmike) [inline docs/faq.rst]: What do you think about splitting this in two sections - "Website is not rendered correctly" and "Splash is crashing"? currently both crashing and wrong rendering is under "wrong rendering"

File: docs/faq.rst [hunk 2]
```diff
@@ -300,6 +292,43 @@ Common reasons:
   A quick (though not precise) way to check it is to try opening a page
   in Safari.
 
+**Splash crashes.**
+
+Common reasons:
+
+* Qt or WebKit bugs which cause Splash to hang or crash. Often the whole
+  website works, but some specific .js (or other) file causes problems.
+  In this case you can try this steps:
```
Comment (kmike) [inline docs/faq.rst]: this -> these

File: docs/faq.rst [hunk 3]
```diff
@@ -300,6 +292,43 @@ Common reasons:
...
+  * If Splash instance failed and stopped - go to the next steps. For example the last request was to the url like ``https://.../static/tagtag.min.js`` with JS,  and we can suspect that probably it caused the issue.
+  * Try to filter out this script with using :ref:`splash-on-request` with this template (or use use :ref:`request filters` to filter it out) :
+  
+    .. code-block:: lua
+
+     function main(splash, args)
+     splash:on_request(function(request)
+       if request.url:find('tagtag') ~= nil then
+             request:abort()
+         end
```
Comment (kmike) [inline docs/faq.rst]: it seems indentation is a bit off

Comment (kmike) [reply docs/faq.rst]: it'd be also good to use consistent indentation - here 4 spaces are used, while in return statement 2 spaces are used

**Comment by kmike** [comment]
Thanks @PyExplorer! It looks great overall, much more readable than before.

**Comment by PyExplorer** [comment]
Thanks for comments! Will fix them.

File: docs/faq.rst [hunk 4]
```diff
@@ -300,6 +292,45 @@ Common reasons:
...
+                request:abort()
+            end
+         end)
+         assert(splash:go(args.url))
+         assert(splash:wait(0.5))
+         return {
+            html = splash:html(),
+            png = splash:png(),
+            har = splash:har(),
+         }
```
Comment (kmike) [inline docs/faq.rst]: Could you please check indentation again? There are places in this function where 3 spaces are used, and places where 4 spaces are used. I think in this example it should be 4 spaces everywhere.

**Comment by kmike** [comment]
Thanks @PyExplorer!

**Pull request merged**